src/electromagnetics.md

```diff
@@ -478,5 +478,85 @@ to specify these regions. To change these, search the code for `std::map<int, do
 integer attribute to the floating-point material value.
 
 
+## Particle Trajectory Calculation
+
+Particle trajectory calculations are an important part of the Particle-In-Cell
+method but they can also provide an illuminating view of electromagnetic fields
+which can otherwise be difficult to visualize.
+
+### Lorentz Mini Application
+
+The Lorentz mini application approximates the trajectory of an individual
+particle using the explicit Boris algorithm. The equations being solved track
+the position and momentum of a particle with a given mass and charge.
+
+$$\begin{align}
+  \frac{d\vec{x}}{dt} & = \frac{\vec{p}}{m} \\\\
+  \frac{d\vec{p}}{dt} & =
+  q\left(\E + \frac{1}{m}\vec{p}\times\B\right)
+\end{align}$$
+
+The Boris algorithm, while not a symplectic method, provides long term accuracy
+by conserving volume in phase space. The algorithm updates the position in a single step but splits the momentum update into multiple steps:
+
+$$\begin{align}
+  \vec{p}\_- & = \vec{p}\_k + \frac{q \Delta t}{2} \E\_k \\\\
+  \vec{p}\_+ - \vec{p}\_- & =
+  \frac{q \Delta t}{2 m} \left(\vec{p}\_+ + \vec{p}\_-\right)\times\B\_k
+  \label{boris2} \\\\
+  \vec{p}\_{k+1} & = \vec{p}\_+ + \frac{q \Delta t}{2} \E\_k \\\\
+  \vec{x}\_{k+1} & = \vec{x}\_k + \frac{\Delta t}{m} \vec{p}\_{k+1}
+\end{align}$$
+
+While equation \eqref{boris2} may appear to be implicit it is clearly a linear
+equation so a small amount of algebra will produce an explicit, though more
+messy and less intuitive, formula.
+
+[![](img/examples/lorentz.png)](img/examples/lorentz.png)
+
+The above image shows a trajectory computed by the Lorentz mini application
+for a charged particle in the presence of a strong bar magnet and a relatively
+weak charged sphere. The trajectory exhibits the magnetic mirror effect where
+the higher magnetic field density near the poles causes the helical orbits to
+tighten and eventually reflect back towards the opposite pole.
+
+#### Mini Application Features
+
+**Electric Field:** The electric field is assumed to be zero unless a
+  VisItDataCollection is provided. The command line option `-er` can be used
```

Comment:
```suggestion
  VisIt data collection is provided. The command line option `-er` can be used
```
The user does not provide the MFEM object, but path to data, so should be rather written like this (and also we do not need to care about formatting of the class name)